diesel/src/connection/mod.rs

```diff
@@ -78,11 +78,11 @@ pub trait Connection: SimpleConnection + Sized {
         U: Queryable<T::SqlType, Self::Backend>,
     {
         self.query_all(source)
-            .and_then(|mut e| e.nth(0).map(Ok).unwrap_or(Err(Error::NotFound)))
+            .and_then(|e: Vec<U>| e.into_iter().next().ok_or(Error::NotFound))
```

Comment:
Should this be `.and_then(|e| e.first().map(Ok).unwrap_or(Err(Error::NotFound)))`?